### ISSUE TITLE
Amethyst game benchmark

### ISSUE THREAD

**Issue by erlend-sh** [katharostech/amethyst-ecs-benchmarks#2]
One of the goals of this benchmark will be to make a [real game as a benchmark](https://community.amethyst.rs/t/legion-ecs-discussion/965/7?u=erlend_sh). Essentially a stripped down version of Evoli, without much of any game logic.

An easier starting point might be something like @maikklein’s https://github.com/MaikKlein/bench_defense, made for [pyro](https://github.com/MaikKlein/pyro). But instead of 2D assets we use 3D (to cover 3D culling testing), and time control UI can be extracted out from Evoli.

I guess you’re waiting for OpenGL support to land before you can work with Amethyst >0.10?

**Comment by zicklag** [comment]
Yes, I am waiting on OpenGL support before I can run any graphics related tests on Amethyst > 0.10, but it looks like that is comming fairly soon. Also, before we can actually run a full Amethyst game as a benchmark, we would have to port any required Amethyst systems to use Legion instead of Specs. I'm starting with the Transform system, that should let us get some decent benchmarks going. After that we might be able to work on the physics integration.

If we got a Specs API wrapper around Legion, we might be able to get the whole of Amethyst running on Legion, but otherwise I think we will have to be limited to a subset of Amethyst features for the benchmarking.
